### PR TITLE
feat(donate-block): support modal checkout

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Support the use of a modal checkout for https://github.com/Automattic/newspack-blocks/pull/1355

Further details and instructions at https://github.com/Automattic/newspack-blocks/pull/1355

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by dkoo — APPROVED** [review]

File: includes/class-donations.php [hunk 1]
```diff
@@ -873,6 +888,24 @@ public static function can_use_streamlined_donate_block() {
...
+	 * Whether the WC cart contains a donation product.
+	 *
+	 * @return bool
+	 */
+	public static function is_donation_cart() {
+		if ( ! self::is_platform_wc() ) {
+			return false;
+		}
+		$donation_products_ids = array_values( self::get_donation_product_child_products_ids() );
+		foreach ( WC()->cart->cart_contents as $prod_in_cart ) {
```
Comment (dkoo) [inline includes/class-donations.php:901]: Getting two PHP warnings on this line when visiting the Reader Revenue wizard:

```
PHP Notice:  Trying to get property 'cart_contents' of non-object in /newspack/app/public/wp-content/plugins/newspack-plugin/includes/class-donations.php on line 901
PHP Warning:  Invalid argument supplied for foreach() in /newspack/app/public/wp-content/plugins/newspack-plugin/includes/class-donations.php on line 901
```

File: includes/wizards/class-reader-revenue-wizard.php [hunk 2]
```diff
@@ -433,19 +433,29 @@ public function fetch_all_data() {
 		$stripe_data                            = Stripe_Connection::get_stripe_data();
 		$stripe_data['can_use_stripe_platform'] = Donations::can_use_stripe_platform();
 
+		$billing_fields = [];
+		if ( 'wc' === $platform && $wc_installed ) {
```
Comment (dkoo) [inline includes/wizards/class-reader-revenue-wizard.php:437]: Nit: There's a helper method `Donations::is_platform_wc()` which you could use here.

**Review by dkoo — CHANGES_REQUESTED** [review]
Some minor PHP warning issues noted below, and maybe we want a quick design review by Thomas or Dani on the checkbox UI, but it works for me.

File: includes/wizards/class-reader-revenue-wizard.php (hunk 2 above)
Comment (miguelpeixe) [reply includes/wizards/class-reader-revenue-wizard.php]: Fixed with 0a6a4fc66a093394e052e076badfe656f8570282

File: includes/class-donations.php (hunk 1 above)
Comment (miguelpeixe) [reply includes/class-donations.php:907]: Thank you for catching that! fd1b2acf5c35642eabc2ce494499a0c577e205c2 ensures there are product and a cart before iterating.

Comment (dkoo) [reply includes/class-donations.php:908]: No longer seeing the warnings, thanks!

**Review by dkoo — APPROVED** [review]
Working well now, and can confirm that the default billing fields now match Woo's full set of default billing fields.

**Pull request merged**